### PR TITLE
Dexterity Implementation of ftw.subsite

### PR DESCRIPTION
**Pull request by maethu**
This PR provides the following features:

- [x] Complete replacement of the Archetype based content.
- [x] Use ftw.simplelayout as startpage.
- [x] Use collective.z3form.widgets for referencing Subsites langauages.
- [x] In-place migration (AT -> DX) using base classes of plone.app.contenttypes 1.1b5 (seems to be stable)
- [x] Migration of Teaser and StaticText Portles to TextBlocks.

The ftw.subsite 2.0.x will provide this migration, thus the old subsite_view is still accessible and functional. ftw.subsite 2.1.x will completely remove all traces of the old AT based Subsite.

### CODE REVIEW

**Comment by maethu** [comment]
@mbaechtold your turn :smile: :wink: :stuck_out_tongue_closed_eyes:

**Comment by maethu** [comment]
@mbaechtold Tests are now passing :smile:

**Comment by mbaechtold** [comment]
@maethu, could you please update the translations? There are 2 or 3 new untranslated message strings.

File: docs/HISTORY.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 Changelog
 =========
 
-1.4.3 (unreleased)
+2.0.0 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Implement Dexterity based subsite content.
```
Comment (mbaechtold) [inline docs/HISTORY.txt]: Maybe you could add your name and a hint about the migration?

File: ftw/subsite/configure.zcml [hunk 2]
```diff
@@ -32,10 +34,12 @@
       provides="Products.GenericSetup.interfaces.EXTENSION"
       />
 
- <subscriber
-      for=".interfaces.ISubsite
-           Products.Archetypes.interfaces.IObjectInitializedEvent"
-      handler=".handlers.block_parent_portlets" />
+  <include package="ftw.upgrade" file="meta.zcml" />
```
Comment (mbaechtold) [inline ftw/subsite/configure.zcml]: A trick I learned from @jone:

Import the upgrades here by including the package with `<include package=".upgrades" />`, then configure the upgrades directory in `ftw/subsite/upgrades/configure.zcml` *after* the existing upgrade steps:

``` 
<upgrade-step:directory
    profile="ftw.subsite:default"
    directory="."
/>
```

This way the old upgrade steps are displayed too.

---

Before:

vorher

---

After:

nachher

---

File: README.rst [hunk 3]
```diff
@@ -39,15 +39,27 @@ Additional functionality provided by `ftw.subsite`
 Compatibility
 -------------
 
-Plone 4.3
+Plone 4.3 + Dexterity + Migration
+
+With ``ftw.subsite`` 2.x only the AT based Subsite content is replaced with a
+Dexterity based Subsite content + ftw.simplelayout default page. Thus the
+Subsite portlet column, column renderer and the teaser portlet are no longer used and will be removed with ``ftw.subsite`` 2.1.0. The ``ftw.subsite`` release 2.0.x will provide a inplace migration. Be aware only static text portlets and subsite teaser portlets will be migrated. You need to provide your own migration for other portlets.
+The subsite_view will be available in ``ftw.subsite`` 2.0.x and removed in 2.1.x.
```
Comment (mbaechtold) [inline README.rst]: As discussed with @maethu this morning: please add a note about the compatibility and the requirements (`plone.app.contenttypes` and `plone.app.event`).

File: ftw/subsite/upgrades/20150622153850_migrate_to_dexterity/upgrade.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from ftw.subsite.migration.mgiration import SubsiteMigration
```
Comment (mbaechtold) [inline ftw/subsite/upgrades/20150622153850_migrate_to_dexterity/upgrade.py]: There is a typo in the modules' name ("mgiration" instead of "migration").

**Comment by mbaechtold** [comment]
I have tested this locally and it worked (the migration too).

**Comment by maethu** [comment]
@mbaechtold Thanks for the feedback - I updated everything you mentioned.

**Comment by maethu** [comment]
@mbaechtold Seems like the hooks are not delivered --> Check https://jenkins.4teamwork.ch/view/All/job/ftw.subsite-dexterity-test-plone-4.3.x.cfg/6/console

File: README.rst [hunk 5]
```diff
@@ -39,15 +39,36 @@ Additional functionality provided by `ftw.subsite`
...
+
+Note about versions:
+
+This package provides a version.cfg. Please make sure you are using suitable versions of `plone.app.contenttypes` and `plone.app.event`.
+
+Currently it's recommend to use:
+
+- plone.app.contenttypes 1.1b5: This is the latest 1.x release. 1.2.x is for Plone 5 and does not work.
+- plone.app.event: 1.1.x supports Plone 4.3. Newer releases will require plone.app.widget, which als tries to install Plone 5.
```
Comment (mbaechtold) [inline README.rst]: s/als/also

**Pull request merged**